ZichanYang324/BF-Employee-HR
# BGP-11-19-27-28 [Frontend and Backend] visa status management page

# Description
This frontend PR is related to the BGP-11-19-27-28.

## Main changes explained:
- Update DocumentStatus.js and VisaStatusManagement.js to include logic for handling document upload in a specific order and displaying appropriate messages based on the document's current status.
- Create DownloadTemplates.js to provide users with the templates for required documents.
- Add new Redux thunks in visaStatusSlice.js for fetching document status and uploading documents, with proper error handling and feedback to the user.

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` to run this PR locally
3. Navigate to the Visa Status Management page and attempt to upload documents in the specified order or not.


## Screenshots or videos of changes or expecting result:
![image](https://github.com/ZichanYang324/BF-Employee-HR/assets/113407918/3dbc6dc3-24ad-4dae-960c-888d428d96dd)



## Note:
1. make sure log in and then use the link in nav bar to visa status management page
2. you can change the status of document status in mongoDB directly or use post or patch request.
